core/src/Threads/Kokkos_ThreadsExec_base.cpp

```diff
@@ -96,140 +92,29 @@ void* internal_pthread_driver(void*) {
 // Spawn a thread
 
 bool ThreadsExec::spawn() {
-  bool result = false;
-
-  pthread_attr_t attr;
-
-  if (0 == pthread_attr_init(&attr) &&
-      0 == pthread_attr_setscope(&attr, PTHREAD_SCOPE_SYSTEM) &&
-      0 == pthread_attr_setdetachstate(&attr, PTHREAD_CREATE_DETACHED)) {
-    pthread_t pt;
-
-    result = 0 == pthread_create(&pt, &attr, internal_pthread_driver, nullptr);
-  }
+  std::thread t(internal_cppthread_driver);
+  t.detach();
 
-  pthread_attr_destroy(&attr);
-
-  return result;
+  return true;
```

Comment:
Should we still be returning anything from the function then?

Reply:
I guess not. I was being minimalist in my changes, but I could simplify that